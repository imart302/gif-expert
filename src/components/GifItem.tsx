import React, { useEffect } from 'react'

interface GifItemProps {
  title: string,
  url: string,
}

export default function GifItem(props: GifItemProps) {
  
  useEffect(() => {
    console.log(props.url);
  
  }, [])
  
  return (

    <div className='card'>
      <img src={props.url} alt={props.title} />
      <p>{props.title}</p>
    </div>
  )
}
