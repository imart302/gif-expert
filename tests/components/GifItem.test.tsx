import { render, screen } from '@testing-library/react';
import React from 'react';
import GifItem from '../../src/components/GifItem';

describe('test in <GifItem>', () => {
  const title = 'Saitama';
  const url = 'https://saitama.com/image';

  test('should match snapshot', () => {
    const { container } = render(<GifItem title={title} url={url}></GifItem>);
    expect(container).toMatchSnapshot();
  });

  test('should be placed correctly', () => {
    const { container } = render(<GifItem title={title} url={url}></GifItem>);
    const { src, alt } = screen.getByRole<HTMLImageElement>('img');
   
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });


});
