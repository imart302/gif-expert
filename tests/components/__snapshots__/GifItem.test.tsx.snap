// Vitest Snapshot v1

exports[`test in <GifItem> > should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com/image"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;

exports[`test in <GifItem> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama.com/image"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
